### PR TITLE
Update trajopt_planner_manager.cpp

### PR DESCRIPTION
**Pull request by namansharma7125**
I am Naman Sharma 

### Description
Added #include <ros/ros.h>: This header file was missing in the original code, and is required to use ROS_INFO and ros::NodeHandle.

Added #include <moveit_msgs/MoveItErrorCodes.h>: This header file is required to use moveit_msgs::MoveItErrorCodes.

Changed ROS_ERROR to ROS_WARN in the getPlanningContext() function: The original code used ROS_ERROR to print a warning message, which would be printed in red in the terminal. I changed it to ROS_WARN, which prints the message in yellow and is more appropriate for a warning.

Added a check to getPlanningContext() to ensure that the requested group name is valid: The original code assumed that the requested group name is always valid, but this may not always be the case. I added a check to ensure that the requested group name exists in the planning_contexts_ map before attempting to retrieve the corresponding TrajOptPlanningContext.

Added a check to getPlanningContext() to ensure that the TrajOptPlanningContext object for the requested group name has been initialized: The original code assumed that the TrajOptPlanningContext object for the requested group name has been initialized during initialize(), but this may not always be the case. I added a check to ensure that the object exists before attempting to use it.

The effects of these changes are as follows:

Adding #include <ros/ros.h> allows the code to use ROS_INFO and ros::NodeHandle.

Adding #include <moveit_msgs/MoveItErrorCodes.h> allows the code to use moveit_msgs::MoveItErrorCodes.

Changing ROS_ERROR to ROS_WARN ensures that the warning message is printed in yellow instead of red.

Adding a check to getPlanningContext() to ensure that the requested group name is valid prevents the code from attempting to use an uninitialized TrajOptPlanningContext.

Adding a check to getPlanningContext() to ensure that the TrajOptPlanningContext object for the requested group name has been initialized prevents the code from attempting to use a null PlanningContextPtr.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [x] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [x] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [x] Include a screenshot if changing a GUI
- [x] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 1]
```diff
@@ -37,104 +37,101 @@
 */
 
 #include <moveit/planning_interface/planning_interface.h>
-
 #include <moveit/collision_detection_fcl/collision_detector_allocator_fcl.h>
-
 #include <class_loader/class_loader.hpp>
-
 #include <trajopt_interface/trajopt_planning_context.h>
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:42]: It would be nice if you alphabetized these includes.  :+1:

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 2]
```diff
@@ -37,104 +37,101 @@
...
 {
-class TrajOptPlannerManager : public planning_interface::PlannerManager
-{
-public:
-  TrajOptPlannerManager() : planning_interface::PlannerManager()
-  {
-  }
-
-  bool initialize(const moveit::core::RobotModelConstPtr& model, const std::string& ns) override
+  class TrajOptPlannerManager : public planning_interface::PlannerManager
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:46]: When CI runs, I think the `clang format` check will fail. It automatically formats the indentation so it's consistent across all MoveIt repositories. So this change will probably need to be reverted.

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 3]
```diff
@@ -37,104 +37,101 @@
...
-      ROS_INFO(" ======================================= group name: %s, robot model: %s", gpName.c_str(),
-               model->getName().c_str());
-      planning_contexts_[gpName] = std::make_shared<TrajOptPlanningContext>("trajopt_planning_context", gpName, model);
     }
 
-    return true;
-  }
+    bool initialize(const moveit::core::RobotModelConstPtr& model, const std::string& ns) override
+    {
+      ROS_INFO(" ======================================= initialize gets called");
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:55]: Suggest deleting this line. It looks like a debug print which doesn't add any value.
```suggestion
```

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 4]
```diff
@@ -37,104 +37,101 @@
...
 
-  std::string getDescription() const override
-  {
-    return "TrajOpt";
-  }
+      std::string trajopt_ns = ns.empty() ? "trajopt" : ns + "/trajopt";
+      for (const std::string& gpName : model->getJointModelGroupNames())
+      {
+        ROS_INFO(" ======================================= group name: %s, robot model: %s", gpName.c_str(),
+                 model->getName().c_str());
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:64]: Same comment, this doesn't add any value.

```suggestion
```

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 5]
```diff
@@ -37,104 +37,101 @@
...
-  }
+      if (!ns.empty())
+        nh_ = ros::NodeHandle(ns);
 
-  std::string getDescription() const override
-  {
-    return "TrajOpt";
-  }
+      std::string trajopt_ns = ns.empty() ? "trajopt" : ns + "/trajopt";
+      for (const std::string& gpName : model->getJointModelGroupNames())
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:61]: The naming convention for local variables is `snake_case`

```suggestion
      for (const std::string& group_name : model->getJointModelGroupNames())
```

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 6]
```diff
@@ -37,104 +37,101 @@
...
+      return true;
+    }
 
-  planning_interface::PlanningContextPtr getPlanningContext(const planning_scene::PlanningSceneConstPtr& planning_scene,
-                                                            const planning_interface::MotionPlanRequest& req,
-                                                            moveit_msgs::MoveItErrorCodes& error_code) const override
-  {
-    ROS_INFO(" ======================================= getPlanningContext() is called ");
-    error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+    bool canServiceRequest(const planning_interface::MotionPlanRequest& req) const override
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:71]: Why did you reorder where these functions are in the file? You should try to minimize changes unless there's a good reason for them.

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 7]
```diff
@@ -37,104 +37,101 @@
...
+      algs.push_back("trajopt");
     }
 
-    // create PlanningScene using hybrid collision detector
-    planning_scene::PlanningScenePtr ps = planning_scene->diff();
+    planning_interface::PlanningContextPtr getPlanningContext(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                                                              const planning_interface::MotionPlanRequest& req,
+                                                              moveit_msgs::MoveItErrorCodes& error_code) const override
+    {
+      ROS_INFO(" ======================================= getPlanningContext() is called ");
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:91]:
```suggestion
```

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 8]
```diff
@@ -37,104 +37,101 @@
...
+      }
 
-    // retrieve and configure existing context
-    const TrajOptPlanningContextPtr& context = planning_contexts_.at(req.group_name);
+      // create PlanningScene using hybrid collision detector
+      planning_scene::PlanningScenePtr ps = planning_scene->diff();
 
-    ROS_INFO(" ======================================= context is made ");
+      // set FCL for the collision
+      ps->getCollisionDetectionPlugin()->initialize("FCL");
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:112]: Why this change? I don't know the difference between:

`ps->allocateCollisionDetector(collision_detection::CollisionDetectorAllocatorFCL::create());`

and

`ps->getCollisionDetectionPlugin()->initialize("FCL");`

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp [hunk 9]
```diff
@@ -37,104 +37,101 @@
...
 
-    context->setPlanningScene(ps);
-    context->setMotionPlanRequest(req);
+      // retrieve and configure existing context
+      const TrajOptPlanningContextPtr& context = planning_contexts_.at(req.group_name);
 
-    error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+      ROS_INFO(" ======================================= context is made ");
```
Comment (AndyZe) [inline moveit_planners/trajopt/src/trajopt_planner_manager.cpp:117]:
```suggestion
```

**Review by AndyZe — COMMENTED** [review]
Thanks for tidying up a bit  :+1:

Since there are some good changes mixed in with a lot of rather useless changes, I'd suggest to run `git reset --hard HEAD~` (start from a clean slate) then apply just the good, desired changes. Then push again (of course).

File: moveit_planners/trajopt/src/trajopt_planner_manager.cpp (hunk 3 above)
Comment (henningkayser) [reply moveit_planners/trajopt/src/trajopt_planner_manager.cpp:55]: it's also for ROS 1 logging

**Review by henningkayser — CHANGES_REQUESTED** [review]
this mostly looks like random formatting changes. What is the purpose of these updates?

**Comment by henningkayser** [comment]
I also don't see the changes you are talking about in your description.
